modules/frontend/searchsharding.go

```diff
@@ -289,18 +342,90 @@ func (s *searchSharder) shardedRequests(ctx context.Context, metas []*backend.Bl
 			subR := parent.Clone(ctx)
 			subR.Header.Set(user.OrgIDHeaderName, tenantID)
 
-			q := subR.URL.Query()
-			q.Add(api.URLParamBlockID, blockID)
-			q.Add(api.URLParamStartPage, strconv.Itoa(startPage))
-			q.Add(api.URLParamTotalPages, strconv.Itoa(pagesPerQuery))
+			subR, err := api.BuildSearchBlockRequest(subR, &tempopb.SearchBlockRequest{
+				BlockID:       blockID,
+				StartPage:     uint32(startPage),
+				TotalPages:    uint32(pagesPerQuery),
```

Comment:
**nit** The mismatch between these names is a little confusing.  `TotalPages` seems like the total number of pages in the block, but it is actually the number of pages for the query to inspect.  Might think about switching to Start/End page numbers, which conceptually lines up well with start/end times, or renaming to something like `PageCount`

Reply:
Renamed to `pagesToSearch`